utils/fetchItemDetails.ts

```diff
@@ -10,20 +11,24 @@ export default async function getItemDetails(
     .select(column_name as '*')
     .eq('id', user_id)
 
-  const itemIds: number[] = data?.[0][`${column_name}`] ?? []
-
-  if (itemIds) {
-    const itemDetails = await Promise.all(
-      itemIds &&
-        itemIds.map(async item_id => {
-          const { data } = await supabase
-            .from('items')
-            .select('*')
-            .eq('item_id', item_id)
-          return data && data[0]
-        }),
-    )
-    return itemDetails
+  if (!data || data.length === 0) {
+    return []
   }
-  return []
+
+  const itemIds = data[0][`${column_name}`] ?? []
+
+  const fetchedItems = await Promise.all(
+    itemIds.map(async item_id => {
+      const { data } = await supabase
+        .from('items')
+        .select('*')
+        .eq('item_id', item_id)
+      return data?.[0]
+    }),
+  )
+
+  const items = fetchedItems.filter(
+    (item: Item | undefined): item is Item => item !== undefined,
```

Comment:
this here is some pure typescript magic called a 'type predicate'.  the `item is Item` bit means 'what I'm going to return from this little fn is of type `Item`', and this does away with the annoying (Item | undefined)[] situation you were in before.  Just using the filter without this type predicate magic annoyingly doesn't convince Typescript that you've got rid of the undefineds (even though you have)